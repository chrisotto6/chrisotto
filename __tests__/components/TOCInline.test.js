import { render } from '@testing-library/react'
import TOCInline from '@/components/TOCInline'

it('renders TOCInline unchanged', () => {
  const { container } = render(
    <TOCInline
      toc={[
        {
          depth: 2,
          url: '#overview',
          value: 'Overview',
        },
      ]}
      exclude="Overview"
      toHeading={2}
    />
  )
  expect(container).toMatchSnapshot()
})

it('renders TOCInline with exclude array unchanged', () => {
  const { container } = render(
    <TOCInline
      toc={[
        {
          depth: 2,
          url: '#overview',
          value: 'Overview',
        },
        {
          depth: 2,
          url: '#theme-colors',
          value: 'Theme colors',
        },
      ]}
      exclude={[]}
      toHeading={2}
    />
  )
  expect(container).toMatchSnapshot()
})

it('renders TOCInline with asDisclosure', () => {
  const { container } = render(
    <TOCInline
      toc={[
        {
          depth: 2,
          url: '#overview',
          value: 'Overview',
        },
      ]}
      toHeading={2}
      asDisclosure={true}
    />
  )
  expect(container).toMatchSnapshot()
})
