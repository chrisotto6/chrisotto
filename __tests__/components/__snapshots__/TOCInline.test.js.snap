// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders TOCInline unchanged 1`] = `
<div>
  <ul />
</div>
`;

exports[`renders TOCInline with asDisclosure 1`] = `
<div>
  <details
    open=""
  >
    <summary
      class="text-l ml-6 pt-2 pb-2 font-bold"
    >
      Table of Contents
    </summary>
    <div
      class="ml-6"
    >
      <ul>
        <li
          class="false"
        >
          <a
            href="#overview"
          >
            Overview
          </a>
        </li>
      </ul>
    </div>
  </details>
</div>
`;

exports[`renders TOCInline with exclude array unchanged 1`] = `
<div>
  <ul>
    <li
      class="false"
    >
      <a
        href="#overview"
      >
        Overview
      </a>
    </li>
    <li
      class="false"
    >
      <a
        href="#theme-colors"
      >
        Theme colors
      </a>
    </li>
  </ul>
</div>
`;
